### PR TITLE
ci: added static code analysis with Codechecker

### PR DESCRIPTION
**Pull request by wallinux**
CodeChecker is a static analysis infrastructure built on the LLVM/Clang Static Analyzer toolchain.
Using Github action to build net-snmp with SCA.
The Github action Codechecker workflow is started manually.
The result is stored in an artifact with HTML files that can be downloaded
and analyzed.

Reference:
- https://github.com/Ericsson/codechecker

Signed-off-by: Anders Wallin <wallinux@gmail.com>

### CODE REVIEW

File: .github/workflows/buildtest.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
...-features --with-perl-modules --with-default-snmp-version=2 --with-systemd --enable-blumenthal-aes --with-python-modules '--with-transports=AAL5PVC IPX STD UDPshared DTLSUDP TLSTCP' '--with-security-modules=usm tsm ksm' '--with-mib-modules=examples/data_set examples/delayed_instance examples/example examples/notification examples/scalar_int examples/ucdDemoPublic examples/watched smux deliver/deliverByNotify disman/event disman/mteEventNotificationTable disman/mteEventTable disman/mteObjectsTable disman/mteTriggerBooleanTable disman/mteTriggerDeltaTable disman/mteTriggerExistenceTable disman/mteTriggerTable disman/mteTriggerThresholdTable disman/nslookup-mib disman/ping-mib disman/schedule disman/traceroute-mib etherlike-mib examples/netSnmpHostsTable hardware/cpu hardware/fsys hardware/memory host ip-forward-mib ip-mib/inetNetToMediaTable ip-mib/ipDefaultRouterTable ip-mib/ipv4InterfaceTable ip-mib/ipv6InterfaceTable ip-mib/ipv6ScopeZoneIndexTable mibII/ipAddr mibII/mta_sendmail misc/ipfwacc sctp-mib snmp-notification-mib tcp-mib testhandler tunnel ucd-snmp/diskio hardware/sensors ucd-snmp/lmsensorsMib ucd-snmp/extensible udp-mib rmon-mib snmp-usm-dh-objects-mib tlstm-mib tsm-mib'
```
Comment (bvanassche) [inline .github/workflows/buildtest.yml:19]: Has it been considered to use the ci/net-snmp-configure script instead?

File: .github/workflows/codechecker.yml [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
...-features --with-perl-modules --with-default-snmp-version=2 --with-systemd --enable-blumenthal-aes --with-python-modules '--with-transports=AAL5PVC IPX STD UDPshared DTLSUDP TLSTCP' '--with-security-modules=usm tsm ksm' '--with-mib-modules=examples/data_set examples/delayed_instance examples/example examples/notification examples/scalar_int examples/ucdDemoPublic examples/watched smux deliver/deliverByNotify disman/event disman/mteEventNotificationTable disman/mteEventTable disman/mteObjectsTable disman/mteTriggerBooleanTable disman/mteTriggerDeltaTable disman/mteTriggerExistenceTable disman/mteTriggerTable disman/mteTriggerThresholdTable disman/nslookup-mib disman/ping-mib disman/schedule disman/traceroute-mib etherlike-mib examples/netSnmpHostsTable hardware/cpu hardware/fsys hardware/memory host ip-forward-mib ip-mib/inetNetToMediaTable ip-mib/ipDefaultRouterTable ip-mib/ipv4InterfaceTable ip-mib/ipv6InterfaceTable ip-mib/ipv6ScopeZoneIndexTable mibII/ipAddr mibII/mta_sendmail misc/ipfwacc sctp-mib snmp-notification-mib tcp-mib testhandler tunnel ucd-snmp/diskio hardware/sensors ucd-snmp/lmsensorsMib ucd-snmp/extensible udp-mib rmon-mib snmp-usm-dh-objects-mib tlstm-mib tsm-mib'
```
Comment (bvanassche) [inline .github/workflows/codechecker.yml:50]: Same comment here: has it been considered to use ci/net-snmp-configure instead?

**Comment by wallinux** [comment]
I will update it to use the ci/net-snmp-configure instead

Anders Wallin

**Comment by bvanassche** [comment]
This pull request has been merged manually because I wanted to combine the two commits into a single commit. Anyway, thanks for your contribution!

**Comment by bvanassche** [comment]
The code checker is no longer working unfortunately. If I try to run it github reports that the python-virtualenv package is no longer available. Do you perhaps have the time to look into this?

**Pull request closed without merging**